GHzDACs/JumpTableIntro.md

```diff
@@ -0,0 +1,152 @@
+# Introduction
+This is a beginner's documentation for the jump table version of the DACs.
+
+*Remember: 8 bits to one byte, 4 bytes to a word, 256 words to a derp.*
+
+The word "derp" is used instead of "page" because "page" used to refer to a completely unrelated construct in how the FPGA server handled pipelining. That said, "derp" isn't very descriptive and should probably be changed to something like "sentence" or "paragraph".
+
+You can think of the DACs as serving one basic function: you give it a (digital) waveform, and then it plays it out (analog). The pre-jump table version of the DACs had two separate types of commands you could issue: "SRAM write", which defines the waveform (in the SRAM memory); and "memory", which defines the control sequences that play the SRAM. The memory commands were limited to fairly basic operations, such as start, stop, repeat, and wait--it was not possible to have branching code.
+
+The jump table is a replacement for the memory commands that allows for much more complex control. The jump table commands form their own mini-programming language, which contains branching (if-thens) and subroutines (in the form of jumps), as well as the ability to read output from the ADC on the daisy chain.
+
+There are now two pointers that can be manipulated:
+
+* The SRAM pointer determines what the DACs are outputting.
+  * This holds a memory address. The DACs play what is at this address, and the pointer gets incremented sequentially.
+* The jump table pointer holds the index of the currently active jump table operation.
+  * A jump table operation has a few parts:
+    * The operation code defines the type of command: e.g. IDLE, CHECK, CYCLE, JUMP, NOP, END
+    * The fromAddress is an SRAM address. When the SRAM pointer reaches the fromAddress, this jump table operation is called, and will execute on the _next_ cycle (i.e. the SRAM pointer will be at fromAddress+1 when the jump table operation executes).
+    * The toAddress is an SRAM address that is used in the CHECK, CYCLE, and JUMP operations.
+  * Jump table operations can manipulate both the SRAM pointer and the jump table pointer.
+
+# Jump Table Operations
+
+Let's make this concrete by enumerating the allowed jump table operations. Each jump table operation is given by a two byte operation code (opcode), as well as two three-byte SRAM addresses, the fromAddress and toAddress. The first byte of the opcode, Byte 0, gives the command and (possibly) arguments to it. The second byte, Byte 1, is commonly devoted to a jump table index. (NB: everything is little endian. "First byte" means least significant byte.) The encoding used below is copied from John's documentation; for example, `xxjjjjjj` means the first two bits `x` are ignored, while the six bits `j` form a single number, the use of which is given in the description. For details of actually defining the jump table, see the section "Jump Table Write" below.
```

Comment:
It would be helpful here to note that while the byte order is little endian when encoding multi-byte integers in jump table entries, individual bytes are always written with the most-significant bit first, e.g. like regular numbers.